Tools/cases_generator/generate_cases.py

```diff
@@ -428,13 +429,22 @@ def add(name: str) -> None:
             if instr.kind == "op" and instr.is_viable_uop():
                 add(instr.name)
 
-    def write_macro_expansions(self, name: str, parts: MacroParts) -> None:
+    def write_macro_expansions(
+        self, name: str, parts: MacroParts, cache_offset: int
+    ) -> None:
         """Write the macro expansions for a macro-instruction."""
         # TODO: Refactor to share code with write_cody(), is_viaible_uop(), etc.
         offset = 0  # Cache effect offset
         expansions: list[tuple[str, int, int]] = []  # [(name, size, offset), ...]
         for part in parts:
             if isinstance(part, Component):
+                # _PUSH_FRAME is super special; it expands to SAVE_IP(next_instr) + _PUSH_FRAME
+                if part.instr.name == "_PUSH_FRAME":
+                    expansions.append(
+                        ("SAVE_IP", OPARG_SIZES["OPARG_SAVE_IP"], cache_offset)
```

Comment:
Is there an elegant way to just include this in the expansion in `bytecodes.c`? Maybe not, since it's uop-specific...

Reply:
Not really. We'd still have to special-case it here, because the "size" and "oparg" fields in the expansion are still irregular; *and* we'd have to skip it when generating the Tier 1 code for the macro, because we don't want it there.